app/models/course/lesson_plan/item.rb

```diff
@@ -42,9 +42,35 @@ class Course::LessonPlan::Item < ApplicationRecord
     )
   }
 
+  # @!method self.opening_within_next_day
+  #   Scopes the lesson plan items to those which are opening in the next 24 hours.
+  scope :opening_within_next_day, (lambda do
+    where(start_at: (Time.zone.now)..(1.day.from_now))
+  end)
+
   belongs_to :course, inverse_of: :lesson_plan_items
   has_many :todos, class_name: Course::LessonPlan::Todo.name, inverse_of: :item, dependent: :destroy
 
+  # Finds the lesson plan items which are starting within the next day for a given course.
+  # Rearrange the items into a hash keyed by the actable type as a string.
+  # For example:
+  # {
+  #   ActableType_1_as_String => [ActableItems...],
+  #   ActableType_2_as_String => [ActableItems...]
+  # }
+  #
+  # @param course [Course] The course to check for published items starting within the next day.
+  # @return [Hash]
+  def self.upcoming_items_from_course_by_type(course)
+    opening_items = course.lesson_plan_items.published.opening_within_next_day.includes(:actable)
+    opening_items_hash = Hash.new { |hash, actable_type| hash[actable_type] = [] }
```

Comment:
Minor, but we can use the simpler version here, ie. `hsh = Hash.new([])`.

Reply:
This doesn't automatically assign the array to the hash.

The assignment will then have to be done in the loop.

`opening_items_hash[item.actable_type] = opening_items_hash[item.actable_type].push(item.actable)`

Reply:
Actually this doesn't work because it doesn't actually return a new array if the key is not found. It returns the same array with all the existing items.